lib/lz4.c

```diff
@@ -822,6 +829,14 @@ LZ4_FORCE_INLINE int LZ4_compress_generic(
          * - token and *token : position to write 4-bits for match length; higher 4-bits for literal length supposed already written
          */
 
+        if ((outputLimited == fillOutput) &&
+            (op + 2 /* offset */ + 1 /* token */ + MFLIMIT - MINMATCH /* min last literals so last match is <= end - MFLIMIT */ > olimit)) {
```

Comment:
One comment about this test. Technically this is more cautious than it needs to be. You only need to reserve space for `MFLIMIT - MINMATCH` literals at the end in the case that the last match is `MINMATCH` long. For long matches, you only need space for `LASTLITERALS` bytes. However, we don't yet know the length of the match here, and we do have to do some kind of check to make sure it's safe to write the offset, so we can't just move this down to where we know the match length. It's only a couple bytes difference, so I suspect it's not worth trying to engineer a way to be perfect.

Reply:
I fully agree.
That looks good enough to me.